### PR TITLE
improving Domain.join

### PR DESCRIPTION
**Pull request by campospinto**
I started a docstring for Domain.join(), and also added the possibility of passing the patch objects (not just their local list indices) in the connectivity list.

This should fix issue [#140](https://github.com/pyccel/sympde/issues/140)

- [x] some pieces of the docstring need to be further clarified

### CODE REVIEW

File: sympde/topology/domain.py [hunk 1]
```diff
@@ -403,13 +472,21 @@ def join(cls, patches, connectivity, name):
         assert all(p.dim==patches[0].dim for p in patches)
         dim = int(patches[0].dim)
 
+        patch_given_by_indices = (len(connectivity) > 0 and isinstance(connectivity[0][0][0], int))
```
Comment (saidctb) [inline sympde/topology/domain.py:475]: In which case will the patches be given by the connectivity and not be passed as a list?

Comment (campospinto) [reply sympde/topology/domain.py:475]: patches are always passed as a list. we are talking about how the patches are passed in the connectivity argument here: they can be passed
 - either by their indices in the list (as it was done before)
 - or as objects (which allows for a clearer syntax when the function is called: this is the example in the docstring)
I can specify this in the docstring

Comment (campospinto) [reply sympde/topology/domain.py:475]: An important missing information, though, is how the orientation is defined. I could not find this information.

Comment (saidctb) [reply sympde/topology/domain.py:487]: For the orientation you can check the reference specified in the docstring
T. Dokken, E. Quak, V. Skytt. Requirements from Isogeometric Analysis for changes in product design ontologies, 2010.
The orientations are specified in the same manner as in GeoPDES

Comment (campospinto) [reply sympde/topology/domain.py:487]: I just checked this reference but could not find a simple definition of the orientation triple in 3D. We should recall this definition and provide a simple example

Comment (saidctb) [reply sympde/topology/domain.py:487]: [Here](https://github.com/rafavzqz/geopdes/blob/master/geopdes/doc/geo_specs_mp_v21.txt#L193-L237) you can find a better explanation with examples

Comment (campospinto) [reply sympde/topology/domain.py:487]: Thanks, I'm adding this link to the docstring.

Comment (campospinto) [reply sympde/topology/domain.py:487]: @saidctb do you have further questions ?

Comment (saidctb) [reply sympde/topology/domain.py:487]: No, it looks good to me

**Review by saidctb — APPROVED** [review]

**Pull request merged**